lib/diaspora/federation/entities.rb

```diff
@@ -200,37 +177,29 @@ def self.reshare(reshare)
         )
       end
 
-      def self.retraction(target)
+      def self.retraction(retraction)
+        case retraction.data[:target_type]
+        when "Contact"
```

Comment:
Yes, I tried that, but the problem is that all other entities (not `Retraction`) need the object still in the database for `Workers::DeferredDispatch` while Retractions use `Workers::DeferredRetraction` and have all date needed in the job parameters. So we need a special case for contact retractions somewhere, and I think that the current solution was the one with the least specials (compared to create it's own special sidekiq worker), since we can still use `Workers::DeferredRetraction` and also I think it's OK since it is some kind of "retraction".

One other alternative I had was, that we could use `Diaspora::Federation::Dispatcher.build(...).dispatch` instead of `Diaspora::Federation::Dispatcher.defer_dispatch(...)` [here](https://github.com/SuperTux88/diaspora/blob/e994c31590aef3b4afadbec6aa51f7e68440c8cb/app/models/user/connecting.rb#L38) and build the federation-salmon in the request context before deleting the `Contact` in the database, so we don't need the `Contact` in the database afterwards.

What solution do you prefer? Or do you have a third cleaner solution that I didn't think of?